src/lib/tools/currency_converter.py

```
@@ -24,62 +24,72 @@ def __init__(self):
+        if raw_response.status_code != 200:
+            raise CurrencyAPIError(
+                f"Currency API error: {raw_response.status_code} - {raw_response.text}"
+            )
+        return raw_response
+
+    def _parse_response(self, raw_response) -> CurrencyConversionResponse:
+        """Parse API JSON response safely."""
+        try:
+            return CurrencyConversionResponse(**raw_response.json())
+        except ValueError as e:
+            raise CurrencyAPIError(f"Invalid API response format: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)